modules/virt-about-tekton-tasks-operator.adoc

```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// * virt/virt-architecture.adoc
+
+:_content-type: CONCEPT
+[id="virt-about-tekton-tasks-operator_{context}"]
+= About tekton-tasks-operator
+
+The `tekton-tasks-operator` deploys additional Tekton tasks that allow users to create VMs from templates, copy and modify templates, and create DataVolumes.
```

Comment:
The picture does not reflect the text.
@ksimon1 Is the term `Tekton` allowed here, or is the correct name "OpenShift Pipeline tasks" ?

```suggestion
... In addition, the `tekton-tasks-operator` deploys example pipelines showing the usage of OpenShift Pipelines for VMs.
```

Reply:
Since this is openshift documentation, I would use `OpenShift Pipeline tasks`

Reply:
@dominikholler I don't understand what you mean by the comment "The picture does not reflect the text". Please clarify.

Reply:
In images/cnv_components_tekton-tasks-operator.png the tekton-tasks-operator looks like it would deploy nothing, so the additional Tekton tasks that allow users to create VMs from templates, copy and modify templates, and create DataVolumes and the example pipelines showing the usage of OpenShift Pipelines for VMs are missing in images/cnv_components_tekton-tasks-operator.png .

Reply:
@dominikholler I see your suggested sentence change starts with "in addition". Do you want to remove the highlighted sentence, or add on to it

Reply:
@sabrinajess , thanks for noticing. You are right, the suggestion was in addition, not instead:
```suggestion
The `tekton-tasks-operator` deploys additional Tekton tasks that allow users to create VMs from templates, copy and modify templates, and create DataVolumes. In addition, the `tekton-tasks-operator` deploys example pipelines showing the usage of OpenShift Pipelines for VMs.
```

Reply:
@dominikholler I believe the paths as shown came from the graphviz script, and we didn't include all possible deployments like templates and other tasks, I think because it has the potential to get a little messy. As I understand it, the graphics should be more high-level. @orenc1 Is there any reason not to include the additional deployments noted by @dominikholler here and in the comment below, for all aspects? Would that be too complicated or can we make it work? Should we add further details to the diagrams for the Tech Debt post-4.11 epic?

Reply:
@orenc1 tekton-task-operator (TTO) deploys only yaml manifests for example `pipelines` and `tasks` (e.g. https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/tasks/copy-template/manifests/copy-template.yaml), which are referencing these images. The image is downloaded only when user creates `taskRun`. Tekton will then create new pod which is using image (e.g. kubevirt-tekton-tasks-copy-template) as a container.
TTO will deploy `tasks` and `pipelines` only if tekton / openshift pipelines is present in cluster (more precise, tto is looking for `tasks.tekton.dev` crd) and if `spec.featureGates.deployTektonTaskResources` in HCO CR is set to true. If these conditions are not true, TTO will not deploy anything.

Reply:
Understood, thanks for the explanation, @ksimon1.
This explains why these resources were not shown up in the generated graphwiz.
It's up to you to decide if you want to include the list above in the doc's graphic as descendants of `tekton-tasks-operator`.

Reply:
@ksimon1 @orenc1 
We do not have time to update this diagram for 4.11.

In 4.12, we have a task to document the Tekton pipelines and manifests ([CNV-15170](https://issues.redhat.com/browse/CNV-15170)). I think it makes sense to add a task ([CNV-20177](https://issues.redhat.com/browse/CNV-20177)) to update this diagram at that point, even if the Tekton tasks are only deployed when everything is enabled.